Don't ignore empty MO translation files

berarma opened the pull request:
Issue #10964.

berarma committed:
Don't ignore empty MO translation files

markstory commented on src/I18n/MessagesFileLoader.php:
```diff
@@ -126,7 +125,7 @@ public function __invoke()
 
         if (!$file) {
-            return $package;
+            return false;
```
Will consumers of this class be ok with getting `false`? I think that's why the old code returned an empty `Package` instance.

berarma replied:
If that's true then I don't know how to deal with non-existing files. It's perfectly fine for a package to be empty.

markstory replied:
Is the empty package causing the cascade to the PO file?

berarma replied:
Yes. I need some way to differentiate when the file doesn't exists or doesn't have any translations.

markstory commented on tests/TestCase/I18n/MessagesFileLoaderTest.php:
```diff
@@ -64,4 +64,15 @@ public function testCustomLocalePath()
+    {
+        $loader = new MessagesFileLoader('empty', 'es', 'mo');
+        $package = $loader();
+        $this->assertNotNull($package);
```
`false` is not null too. Are there any other assertions we can make about this package object?

berarma replied:
My bad. I should check for false.

berarma commented:
I've changed the assertions in the tests and updated the docblock for the modified method.

It doesn't seem the ```MessagesFileLoader``` class is used anywhere else than the ```TranslationRegistry``` class.

The ```ChainMessagesLoader``` class that invokes the class already does a fuzzy check for a false return value, thus I thought it would OK to use false as return value when the file doesn't exist. If a Package should always be expected then that check would be useless.

I could change the return value to null if that makes more sense.

markstory commented:
I took a look and you're right about the `!$package` check. Given that exists, returning false is reasonable.
